extensions/arc/deployment/src/test/java/io/quarkus/arc/test/unused/UnusedExclusionTest.java

```diff
@@ -21,13 +35,56 @@
     static final QuarkusUnitTest config = new QuarkusUnitTest()
             .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
                     .addClasses(UnusedExclusionTest.class, Alpha.class, Beta.class, Charlie.class, Delta.class,
-                            ProducerBean.class)
+                            ProducerBean.class, TestRecorder.class, Gama.class, GamaProducer.class)
                     .addAsResource(new StringAsset(
                             "quarkus.arc.unremovable-types=io.quarkus.arc.test.unused.UnusedExclusionTest$Alpha,io.quarkus.arc.test.unused.subpackage.**,io.quarkus.arc.test.unused.Charlie,Delta"),
-                            "application.properties"));
+                            "application.properties"))
+            .addBuildChainCustomizer(buildCustomizer());
+
+    static Consumer<BuildChainBuilder> buildCustomizer() {
+        return new Consumer<BuildChainBuilder>() {
+
+            @Override
+            public void accept(BuildChainBuilder builder) {
+                builder.addBuildStep(new BuildStep() {
+
+                    @Override
+                    public void execute(BuildContext context) {
+                        BeanContainer beanContainer = context.consume(BeanContainerBuildItem.class).getValue();
+                        BytecodeRecorderImpl bytecodeRecorder = new BytecodeRecorderImpl(true,
+                                TestRecorder.class.getSimpleName(),
+                                "test", "" + TestRecorder.class.hashCode());
+                        // We need to use reflection due to some class loading problems
+                        Object recorderProxy = bytecodeRecorder.getRecordingProxy(TestRecorder.class);
```

Comment:
Let's keep an eye out for future uses of this and try to move it into some test utility if it starts shows up

Reply:
FYI @michalszynkiewicz